app/code/Magento/UrlRewrite/Model/Storage/DbStorage.php

```diff
@@ -375,4 +375,33 @@ public function deleteByData(array $data)
             $this->prepareSelect($data)->deleteFromSelect($this->resource->getTableName(self::TABLE_NAME))
         );
     }
+
+    /**
+     * Function deleteEntitiesFromStores
+     *
+     * Deletes multiple URL Rewrites from database
+     *
+     * @param array $store_ids
+     * @param array $entity_ids
+     * @param int $entity_type
+     */
+    public function deleteEntitiesFromStores($store_ids, $entity_ids, $entity_type)
```

Comment:
Could you extract this method to separate "service" class? and use something like this:
```
$this->deleteEntitiesFromStores->execute($store_ids, $entity_ids, $entity_type)
```

In addition to that please use `$storeIds`, `$entityIds`, `$entityType` instead, define argument types and return type.

Reply:
Done. I have just moved into own class and renamed. I haven't created any interfaces/API structures. Is that what you were after?